docs/rfcs/fix-2-to-1-spacing.md

```diff
@@ -0,0 +1,49 @@
+
+## Problem
+
+![](../assets/images/2-to-1-spacing-problem.png)
+
+- The bars don’t occupy the entire space due to the default bar width of 16px and a fixed 2:1 spacing, often leaving considerable amount of blank space before the first bar and after the last bar. This becomes particularly noticeable when the chart width is large.
+- Additionally, this setup causes unnecessary overlapping of x-axis labels.
+
+## Proposal
+
+1. Add optional `xAxisInnerPadding`, `xAxisOuterPadding` props to vertical bar charts.
+   - `xAxisInnerPadding` controls the space between bars, while `xAxisOuterPadding` controls the space before the first bar and after the last bar.
+   - These props accept values between 0 and 1, representing a fraction of the [step](https://d3js.org/d3-scale/band#band_step), which is the interval between the start of a bar and the start of the next bar. These props are particularly relevant when using a string x-axis. For additional information on padding in string axes, see [Band scales | D3 by Observable](https://d3js.org/d3-scale/band#band_paddingInner).
+   - By default, the inner padding is set to 2/3, meaning the space between bars is twice the bar width, while the outer padding is set to 1/3, implying that the space before the first bar and after the last bar is equivalent to the bar width. These default values are calculated using the formula:
+     $$
+     innerPadding = {spaceBetweenBars \over spaceBetweenBars + barWidth}
+     $$
+     For a more detailed explanation of how these values were derived, see [Implementing 2:1 spacing | FluentUI Charting Contrib Docsite](https://microsoft.github.io/fluentui-charting-contrib/docs/implementing-2-to-1-spacing).
+   - The default outer padding is non-zero, but users have the option to override it with a value of 0 if they prefer no space before the first bar and after the last bar. Therefore, the minimum domain margin of 8px specified in the design is disregarded for string x-axis.
+2. Use the existing `barWidth` prop and add an optional `maxBarWidth` prop to vertical bar charts.
+   - The `barWidth` prop takes precedence over the props mentioned above. If not explicitly specified, the bar width will be dynamically adjusted to avoid overlapping while maintaining specified spacing between and around the bars. However, when `barWidth` is provided, it will be respected, potentially compromising padding.
+   - In cases where the chart width is large, the bar width might become excessively large. To mitigate this, the `maxBarWidth` prop can be utilized to restrict the width of the bars to a specified maximum.
```

Comment:
Create a decision tree covering the whole logic. Start with - axis type -> numeric or data -> no action. axis type -> string -> follow the order as you have described.

Reply:
done